pkg/server/container_stats_list_windows.go

```diff
@@ -109,13 +109,31 @@ func (c *criService) getSandboxMetrics(
 		}
 		if s, ok := v.(*runhcsstats.Statistics); ok {
 			timestamp := stats.Timestamp.UnixNano()
-			cs.Cpu = &runtime.CpuUsage{
-				Timestamp:            timestamp,
-				UsageCoreNanoSeconds: &runtime.UInt64Value{Value: s.VM.Processor.TotalRuntimeNS},
-			}
-			cs.Memory = &runtime.MemoryUsage{
-				Timestamp:       timestamp,
-				WorkingSetBytes: &runtime.UInt64Value{Value: s.VM.Memory.WorkingSetBytes},
+			// If the pod is process isolated VM will be nil. We have special treatment
+			// for hypervisor isolated pod stats as we return the UVMs working set and
+			// other metrics instead of the actual sandbox containers. This also avoids
+			// a nil pointer dereference on trying to use the VM stats for process isolated.
```

Comment:
nit: I would remove this last sentence. Once this issue is fixed, I don't think it will be important to state that there was once a problem with a nil pointer dereference here.

Reply:
Yea the VM field being nil comment should be enough to indicate this. Will change